openfe/protocols/openmm_afe/equil_solvation_afe_method.py

```diff
@@ -0,0 +1,575 @@
+          being desolvated.
+        """
+
+        # Crash out if there are any alchemical components in state B for now
+        if len(alchemical_components['stateB']) > 0:
+            errmsg = ("Components appearing in state B are not "
+                      "currently supported")
+            raise ValueError(errmsg)
+
+        if len(alchemical_components['stateA']) > 1:
+            errmsg = ("More than one alchemical components is not supported "
+                      "for absolute solvation free energies")
+            raise ValueError(errmsg)
+
+        # Crash out if any of the alchemical components are not
+        # SmallMoleculeComponent
+        for comp in alchemical_components['stateA']:
+            if not isinstance(comp, SmallMoleculeComponent):
+                errmsg = ("Non SmallMoleculeComponent alchemical species "
+                          "are not currently supported")
+                raise ValueError(errmsg)
+
+    def _create(
+        self,
+        stateA: ChemicalSystem,
+        stateB: ChemicalSystem,
+        mapping: Optional[Dict[str, gufe.ComponentMapping]] = None,
+        extends: Optional[gufe.ProtocolDAGResult] = None,
+    ) -> list[gufe.ProtocolUnit]:
+        # TODO: extensions
+        if extends:  # pragma: no-cover
+            raise NotImplementedError("Can't extend simulations yet")
+
+        # Validate components and get alchemical components
+        self._validate_solvent_endstates(stateA, stateB)
+        alchem_comps = system_validation.get_alchemical_components(
+            stateA, stateB,
+        )
+        self._validate_alchemical_components(alchem_comps)
+
+        # Check nonbond & solvent compatibility
+        solv_nonbonded_method = self.settings.solvent_system_settings.nonbonded_method
+        vac_nonbonded_method = self.settings.vacuum_system_settings.nonbonded_method
+        # Use the more complete system validation solvent checks
+        system_validation.validate_solvent(stateA, solv_nonbonded_method)
+        # Gas phase is always gas phase
+        if vac_nonbonded_method.lower() != 'nocutoff':
+            errmsg = ("Only the nocutoff nonbonded_method is supported for "
+                      f"vacuum calculations, {vac_nonbonded_method} was "
+                      "passed")
+            raise ValueError(errmsg)
+
+        # Get the name of the alchemical species
+        alchname = alchem_comps['stateA'][0].name
+
+        # Create list units for vacuum and solvent transforms
+
+        solvent_units = [
+            AbsoluteSolventTransformUnit(
+                stateA=stateA, stateB=stateB,
+                settings=self.settings,
+                alchemical_components=alchem_comps,
+                generation=0, repeat_id=i,
+                name=(f"Absolute Solvation, {alchname} solvent leg: "
+                      f"repeat {i} generation 0"),
+            )
+            for i in range(self.settings.alchemsampler_settings.n_repeats)
```

Comment:
So this is a bit of a weird one conceptually - i.e. is a repeat a full cycle or based on a single leg.

Technically your vacuum legs converge way way faster so you could say "yeah ok you only really need one repeat here".

However, because of the way we propagate errors, and because vacuum is silly cheap to compute, it's a lot easier to assume the former. Conceptually I also prefer looking at full repeats of a cycle since vacuum is like the "correction" to each solvent calculation.